File: java/epi/src/test/java/moe/ijnji/epi/InterconvertingStringIntegerTest.java
package moe.ijnji.epi;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class InterconvertingStringIntegerTest {

    @Test
    public void testSimple() {
        assertEquals(InterconvertingStringInteger.intToString(0), "0");
        assertEquals(InterconvertingStringInteger.intToString(-1), "-1");
        assertEquals(InterconvertingStringInteger.intToString(1), "1");
        assertEquals(InterconvertingStringInteger.intToString(2), "2");
        assertEquals(InterconvertingStringInteger.intToString(-2), "-2");
        assertEquals(InterconvertingStringInteger.intToString(9), "9");
        assertEquals(InterconvertingStringInteger.intToString(10), "10");
        assertEquals(InterconvertingStringInteger.intToString(123), "123");
        assertEquals(InterconvertingStringInteger.intToString(Integer.MAX_VALUE), Integer.toString(Integer.MAX_VALUE));

        assertEquals(InterconvertingStringInteger.stringToInt(""), 0);
        assertEquals(InterconvertingStringInteger.stringToInt("0"), 0);
        assertEquals(InterconvertingStringInteger.stringToInt("-1"), -1);
        assertEquals(InterconvertingStringInteger.stringToInt("1"), 1);
        assertEquals(InterconvertingStringInteger.stringToInt("2"), 2);
        assertEquals(InterconvertingStringInteger.stringToInt("-2"), -2);
        assertEquals(InterconvertingStringInteger.stringToInt("9"), 9);
        assertEquals(InterconvertingStringInteger.stringToInt("10"), 10);
        assertEquals(InterconvertingStringInteger.stringToInt("123"), 123);
        assertEquals(InterconvertingStringInteger.stringToInt(Integer.toString(Integer.MAX_VALUE)), Integer.MAX_VALUE);
    }

}

File: EpiPy/interconverting_string_integer_test.py
#!/usr/bin/env python3

import pytest

from epi.interconverting_string_integer import int_to_string
from epi.interconverting_string_integer import string_to_int

def test_small():
    assert(int_to_string(0) == '0')
    assert(int_to_string(-1) == '-1')
    assert(int_to_string(1) == '1')
    assert(int_to_string(1000) == '1000')
    assert(int_to_string(2147483646) == '2147483646')
    assert(int_to_string(-2147483646) == '-2147483646')
    assert(string_to_int('0') == 0)
    assert(string_to_int('-1') == -1)
    assert(string_to_int('1') == 1)
    assert(string_to_int('1000') == 1000)
    assert(string_to_int('2147483646') == 2147483646)
    assert(string_to_int('-2147483646') == -2147483646)
